tools/log_analyser/main.py

```diff
@@ -0,0 +1,39 @@
+import arranging
+import log_agg
+import open_handles_metric
+import read_pattern_metric
+import count_gcs_calls
+import bytes_transferred
+
+# Add the log files here
+files = []
+itr = ""
+print("Enter the logs file names (with absolute path), press -1 when done:")
+while itr != "-1":
+    itr = input()
+    if itr != "-1":
+        files.append(itr)
+
+print("Entered the time window for which you want the logs to be analysed")
+start_time = int(input("start time(epoch): "))
+end_time = int(input("end time(epoch): "))
+
+file = input("Enter the file name for metrics: ")
```

Comment:
LogFile can be a data classes which can contain filename for now
TimeWindow can be a data class which can contain starttime and endtime
LogAnalysisInput can be a data class to encapsulate all input data (list of log files, time window, and filename
LogAnalysisOutput to encapsulate the output data (total logs and aggregated logs)

Reply:
Initialization: The __init__ method initializes attributes using the LogAnalysisInput object.
Arranging Files: The arrange_files method orders the files.
Aggregating Logs: The aggregate_logs method aggregates the logs using the provided start and end times and filename.
Processing Metrics: The process_metrics method processes various metrics using the respective modules.
Run Method: The run method orchestrates the workflow and returns a LogAnalysisOutput object containing the results.
These can be methods of top level log analysis class

Reply:
Refactoring of the code is done.